libnmstate/nm/applier.py

```diff
@@ -153,6 +156,18 @@ def _set_ifaces_admin_state(ifaces_desired_state, con_profiles):
                     new_ifaces_to_activate.add(ifname)
         else:
             if iface_desired_state[Interface.STATE] == InterfaceState.UP:
+                if iface_desired_state.get(BOND_MODE_CHANGED_METADATA):
```

Comment:
nit: You can hide the use of `BOND_MODE_CHANGED_METADATA` by pushing this condition to the `appliers.bond`.
(e.g. `appliers.bond.is_mode_changed(iface_desired_state)`)